admin/tools/provision/cloudmanager.py

```diff
@@ -0,0 +1,311 @@
+
+BASE_IMAGE_KEY='base_image_name'
+SNAPSHOT_IMAGE_KEY='snapshot_name'
+
+def get_nova_creds():
+    """
+    Extract the login information from the environment
+    """
+    creds = {}
+    creds['version'] = 2
+    creds['username'] = os.environ['OS_USERNAME']
+    creds['api_key'] = os.environ['OS_PASSWORD']
+    creds['auth_url'] = os.environ['OS_AUTH_URL']
+    creds['project_id'] = os.environ['OS_TENANT_NAME']
+    creds['insecure'] = True
+    return creds
+
+def add_parser_args(parser):
+    """
+    Configure the parser
+    """
+    parser.add_argument('-v', '--verbose', dest='verbose', default=[], action='append_const',
+                        const=None, help='More verbose output, can use several times.')
+    parser.add_argument('--verbose-all', dest='verboseAll', default=False, action='store_true',
+                        help='Apply verbosity to all loggers, by default only loader level is set.')
+    # parser = lsst.qserv.admin.logger.add_logfile_opt(parser)
+    group = parser.add_argument_group('Cloud configuration options',
+                                       'Options defining parameters to access remote cloud-platform')
+
+    group.add_argument('-f', '--config', dest='configFile',
+                        required=True, metavar='PATH',
+                        help='Add cloud config file which contains instance characteristics')
+
+    return parser
+
+def config_logger(loggerName, verbose, verboseAll):
+    """
+    Configure the logger
+    """
+    verbosity = len(verbose)
+    levels = {0: logging.WARNING, 1: logging.INFO, 2: logging.DEBUG}
+    if not verboseAll:
+        # suppress INFO/DEBUG regular messages from other loggers
+        # Disable requests and urllib3 package logger and warnings
+        logging.getLogger("requests").setLevel(logging.ERROR)
+        logging.getLogger("urllib3").setLevel(logging.ERROR)
+
+    logging.basicConfig(format='%(asctime)s %(levelname)-8s %(name)-15s'
+                               ' %(message)s',
+                        level=levels.get(verbosity, logging.DEBUG))
+
+    warnings.filterwarnings("ignore")
+
+
+class CloudManager(object):
+    """Application class for common definitions for creation of snapshot and provision qserv"""
+
+    def __init__(self, config_file_name, used_image_key=BASE_IMAGE_KEY, add_ssh_key=False):
+        """
+        Constructor parse all arguments
+
+        @param add_ssh_key Add ssh key only while launching instances in provision qserv.
+        @param go_for_snapshot find new_image until it's created
+        """
+
+        logging.debug("Use configuration file: {}".format(config_file_name))
+
+        config = ConfigParser.ConfigParser({'net-id': None,
+                                            'ssh_security_group': None})
+
+        try:
+            config_file = open(config_file_name, 'r')
+            try:
+                config.readfp(config_file)
+            finally:
+                config_file.close()
```

Comment:
There is a better way to do this using context manager in modern Python, five lines above can be replaced with:

    with open(config_file_name, 'r') as config_file:
        config.readfp(config_file)

or even

    config.readfp(open(config_file_name, 'r'))

to rely on Python garbage collector to close the file.